zipkin/src/main/java/zipkin2/internal/Proto3Fields.java

```diff
@@ -164,6 +164,13 @@ final int readLengthPrefix(Buffer b) {
     static final char[] HEX_DIGITS =
       {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
 
+    // Reuse the buffer for decoding into hex since it's immediately copied into a String.
+    static final ThreadLocal<char[]> THIRTY_TWO_CHARS = new ThreadLocal<char[]>() {
```

Comment:
Sorry for not remembering the specific nuance, but I recall us doing something like this vs using an anonymous subclass in brave. If same to you, mind following that vs us hunting down the conversations around it?

```java
  static final ThreadLocal<char[]> CHAR_BUFFER = new ThreadLocal<>();

  static char[] getCharBuffer() {
    char[] charBuffer = CHAR_BUFFER.get();
    if (charBuffer == null) {
      charBuffer = new char[FORMAT_MAX_LENGTH];
      CHAR_BUFFER.set(charBuffer);
    }
    return charBuffer;
  }
```

Reply:
Sure - it might have something to do with classloader leaks which I'm not familiar with since I've never used a shared servlet container ;)

Reply:
git notifications are delayed for some reason. wires crossed! that was it